include/exadg/functions_and_boundary_conditions/interface_coupling.h

```diff
@@ -533,175 +142,89 @@ class InterfaceCoupling
         }
       }
 
-      map_q_points_src.emplace(quadrature, std::map<mpi_rank, ArrayQuadraturePoints>());
-      map_cache_src.emplace(quadrature, std::map<mpi_rank, ArrayVectorCache>());
-      map_solution_src.emplace(quadrature, std::map<mpi_rank, ArrayVectorTensor>());
-
-      std::map<mpi_rank, ArrayQuadraturePoints> & mpi_map_q_points_src =
-        map_q_points_src.find(quadrature)->second;
-      std::map<mpi_rank, ArrayVectorCache> & mpi_map_cache_src =
-        map_cache_src.find(quadrature)->second;
-      std::map<mpi_rank, ArrayVectorTensor> & mpi_map_solution_src =
-        map_solution_src.find(quadrature)->second;
-
+      array_solution_dst.resize(array_q_points_dst.size(), Tensor<rank, dim, double>());
 
       /*
        * 2. Communication: receive and cache quadrature points of other ranks,
        *    redundantly store own q-points (those that are needed)
        */
```

Comment:
You removed 3. and 4., so we need to make sure that this comment is up-to-date. Please go through your changes from a "comment" point of view.